File: abc/abc176/abc176_c.fs
// https://atcoder.jp/contests/abc176/submissions/16151964

let N = stdin.ReadLine() |> int
let A = stdin.ReadLine().Split() |> Array.map int64

let rec solver s m i =
    if i = N then s
    elif A.[i] < m then solver (s + m - A.[i]) m (i + 1)
    else solver s A.[i] (i + 1)

solver 0L A.[0] 1 |> stdout.WriteLine

File: abc/abc176/abc176_c.cs
// https://atcoder.jp/contests/abc176/submissions/16152722

using System;
using System.Linq;

class Program
{
    static void Main()
    {
        var N = int.Parse(Console.ReadLine());
        var A = Console.ReadLine().Split().Select(int.Parse).ToArray();
        var M = A[0];
        var sum = 0;
        for (int i = 1; i < N; i++)
        {
            if (A[i] < M)
            {
                sum += M - A[i];
            }
            else
            {
                M = A[i];
            }
        }
        Console.WriteLine(sum);
    }
}
